build_environment.bat

```diff
@@ -23,106 +159,154 @@ if %CONFIG_RELEASE% (
   set RUNTIMELIB="MultiThreadedDebugDLL"
   set _CL_=/MDd
 )
-
+SET BUILDTYPEDIR=%2
 if %DYNAMIC_LIBS% (
   echo Building dynamic libraries - DLLs.
   set STATIC_LIBS=0==1
 ) else (
+  SET BUILDTYPEDIR=%2Static
   echo Building static libraries where there's a choice of projects.
   echo    The rest of the project/solution files need to be set to static manually.
   set STATIC_LIBS=1==1
 )
 
 rem Use all CPU cores
-set CL=/MP
-
-rem Edit to suit your environment
-SET MSVC_PATH=C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC
-SET VCVERSION=140
-SET BUILDTOOLS_PATH=E:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\
-SET BUILDTOOLS_SCRIPT=vcvarsall.bat
-rem If using the Visual C++ Build Tools, uncomment & adjust the following
-rem SET BUILDTOOLS_PATH=C:\Program Files (x86)\Microsoft Visual C++ Build Tools
-rem SET BUILDTOOLS_SCRIPT=vcbuildtools.bat
-rem If building for XP and using the C++ Build Tools package (not full VS,)
-rem   install the Windows 7.1 SDK then uncomment and adjust the following
-rem SET WindowsSdkDir_71A=E:\Program Files\Microsoft SDKs\Windows\v7.1\
+set CL=/MP /FS
 
 SET XCOPY=xcopy /S /Y /I
-SET MSBUILD=msbuild /nologo /m /p:PlatformToolset=v%VCVERSION%_xp /p:RuntimeLibrary=%RUNTIMELIB%
-rem If using the Visual C++ Build Tools, uncomment & adjust the following
-rem SET MSBUILD=msbuild /m /p:PlatformToolset=v140_xp /p:VCTargetsPath="C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\\"
+
+set OLDPATH=%PATH%
 set ROOT_DIR=%CD%
-SET BIN_DIR=%CD%\bin\
-SET LIB_DIR=%CD%\lib\
-SET INCLUDE_DIR=%CD%\include\
+SET BIN_DIR=%ROOT_DIR%\%FOLDER_PLATFORM%\%BUILDTYPEDIR%\bin\
```

Comment:
BTW, instead of building all platforms from one clone of the repo, on the build server we clone the repo 4 times (release|debug x86|x64) and build one for each. This prevents the risk of cross-contamination within one environment (which we have screwed up before).